pkg/health/membergarbagecollector/membergarbagecollector.go

```diff
@@ -0,0 +1,173 @@
+	if etcdConfig == nil {
+		return nil, &errors.EtcdError{
+			Message: "nil etcd config passed, can not create heartbeat",
+		}
+	}
+	if clientSet == nil {
+		return nil, &errors.EtcdError{
+			Message: "nil kubernetes clientset passed, can not create heartbeat",
+		}
+	}
+	namespace, err := utils.GetEnvVarOrError(podNamespace)
+	if err != nil {
+		logger.Fatalf("POD_NAMESPACE env var not present: %v", err)
+	}
+	podName, err := utils.GetEnvVarOrError(podName)
+	if err != nil {
+		logger.Fatalf("POD_NAME env var not present: %v", err)
+	}
+	// druidv1alpha1.AddToScheme(scheme.Scheme)
+	return &MemberGarbageCollector{
+		logger:       logger.WithField("actor", "membergc"),
+		etcdConfig:   etcdConfig,
+		k8sClient:    clientSet,
+		podNamespace: namespace,
+		etcdName:     podName[:strings.LastIndex(podName, "-")],
+	}, nil
+}
+
+// RunMemberGarbageCollectorPeriodically periodically calls the member garbage collector
+func RunMemberGarbageCollectorPeriodically(ctx context.Context, hconfig *brtypes.HealthConfig, logger *logrus.Entry, etcdConfig *etcdutil.EtcdConnectionConfig) {
+
+	clientSet, err := miscellaneous.GetKubernetesClientSetOrError()
+	if err != nil {
+		logger.Errorf("failed to create kubernetes clientset: %v", err)
+		return
+	}
+	mgc, err := NewMemberGarbageCollector(logger, etcdConfig, clientSet)
+	if err != nil {
+		logger.Errorf("failed to initialize new heartbeat: %v", err)
+		return
+	}
+	mgc.gcTimer = time.NewTimer(hconfig.MemberGCDuration.Duration)
+	defer mgc.gcTimer.Stop()
+	mgc.logger.Info("Started etcd member garbage collection timer")
+
+	for {
+		select {
+		case <-mgc.gcTimer.C:
+			err := mgc.RemoveSuperfluousMembers(ctx)
+			if err != nil {
+				mgc.logger.Warn(err)
+			}
+			mgc.gcTimer.Reset(hconfig.MemberGCDuration.Duration)
+		case <-ctx.Done():
+			mgc.logger.Info("Stopped etcd member garbage collection timer")
+			return
+		}
+	}
+}
+
+// RemoveSuperfluousMembers removes a member from the etcd cluster if its corresponding pod is not present
+func (mgc *MemberGarbageCollector) RemoveSuperfluousMembers(ctx context.Context) error {
+	etcd := &druidv1alpha1.Etcd{}
+	if err := mgc.k8sClient.Get(ctx, client.ObjectKey{
+		Namespace: mgc.podNamespace,
+		Name:      mgc.etcdName,
+	}, etcd); err != nil {
+		return &errors.EtcdError{
+			Message: fmt.Sprintf("Could not fetch etcd %v with error: %v", mgc.etcdName, err),
+		}
+	}
+	// Return is no members present
+	if len(etcd.Status.Members) < 1 {
+		return &errors.EtcdError{
+			Message: "There are no members present in the etcd status. Nothing to clean",
+		}
+	}
+
+	// Return if memberlist size <= desired cluster size. Cleanup not needed
+	if len(etcd.Status.Members) <= etcd.Spec.Replicas {
+		return nil
+	}
+
+	//Create etcd client to get etcd cluster member list
+	etcdClient, err := etcdutil.GetTLSClientForEtcd(mgc.etcdConfig)
```

Comment:
you have to change the `etcdClient` once [PR#381](https://github.com/gardener/etcd-backup-restore/pull/381) get merged

Reply:
Changed to use `NewCluster`